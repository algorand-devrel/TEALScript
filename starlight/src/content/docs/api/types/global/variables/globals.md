---
editUrl: false
next: false
prev: false
title: "globals"
---

> **`const`** **globals**: `Object`

## Type declaration

### assetCreateMinBalance

> **assetCreateMinBalance**: [`uint64`](../type-aliases/uint64.md)

### assetOptInMinBalance

> **assetOptInMinBalance**: [`uint64`](../type-aliases/uint64.md)

### callerApplicationAddress

> **callerApplicationAddress**: [`Address`](../classes/Address.md)

### callerApplicationID

> **callerApplicationID**: [`Application`](../classes/Application.md)

### creatorAddress

> **creatorAddress**: [`Address`](../classes/Address.md)

### currentApplicationAddress

> **currentApplicationAddress**: [`Address`](../classes/Address.md)

### currentApplicationID

> **currentApplicationID**: [`Application`](../classes/Application.md)

### genesisHash

> **genesisHash**: [`bytes32`](../type-aliases/bytes32.md)

### groupID

> **groupID**: [`bytes32`](../type-aliases/bytes32.md)

### groupSize

> **groupSize**: [`uint64`](../type-aliases/uint64.md)

### latestTimestamp

> **latestTimestamp**: [`uint64`](../type-aliases/uint64.md)

### logicSigVersion

> **logicSigVersion**: [`uint64`](../type-aliases/uint64.md)

### maxTxnLife

> **maxTxnLife**: [`uint64`](../type-aliases/uint64.md)

### minBalance

> **minBalance**: [`uint64`](../type-aliases/uint64.md)

### minTxnFee

> **minTxnFee**: [`uint64`](../type-aliases/uint64.md)

### opcodeBudget

> **opcodeBudget**: [`uint64`](../type-aliases/uint64.md)

### round

> **round**: [`uint64`](../type-aliases/uint64.md)

### zeroAddress

> **zeroAddress**: [`Address`](../classes/Address.md)

## Source

[types/global.d.ts:800](https://github.com/algorandfoundation/tealscript/blob/18ba30a9/types/global.d.ts#L800)
