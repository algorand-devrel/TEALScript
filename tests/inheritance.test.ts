/* eslint-disable func-names */
/* eslint-disable prefer-arrow-callback */
import * as algokit from '@algorandfoundation/algokit-utils';
import { describe, test } from '@jest/globals';
import { artifactsTest, compileAndCreate, runMethod, algodClient, kmdClient } from './common';

const PATH = 'tests/contracts/inheritance.algo.ts';
const ARTIFACTS_DIR = 'tests/contracts/artifacts/';

describe('Inheritance', function () {
  artifactsTest('tests/contracts/inheritance.algo.ts', 'tests/contracts/artifacts/', 'A');
  artifactsTest('tests/contracts/inheritance.algo.ts', 'tests/contracts/artifacts/', 'B');
  artifactsTest('tests/contracts/inheritance.algo.ts', 'tests/contracts/artifacts/', 'C');
  artifactsTest('tests/contracts/inheritance.algo.ts', 'tests/contracts/artifacts/', 'D');
  artifactsTest('tests/contracts/inheritance.algo.ts', 'tests/contracts/artifacts/', 'AD');

  describe('E2E', function () {
    const sender = algokit.getLocalNetDispenserAccount(algodClient, kmdClient);
    test('B extends A', async function () {
      const { appClient } = await compileAndCreate(await sender, PATH, ARTIFACTS_DIR, 'B');
      await runMethod({ appClient, method: 'a', methodArgs: [] });
      await runMethod({ appClient, method: 'b', methodArgs: [] });
    });

    test('C extends B', async function () {
      const { appClient } = await compileAndCreate(await sender, PATH, ARTIFACTS_DIR, 'C');
      await runMethod({ appClient, method: 'a', methodArgs: [] });
      await runMethod({ appClient, method: 'b', methodArgs: [] });
      await runMethod({ appClient, method: 'c', methodArgs: [] });
    });

    test('AD extends Contract.extend(A, D)', async function () {
      const { appClient } = await compileAndCreate(await sender, PATH, ARTIFACTS_DIR, 'AD');
      await runMethod({ appClient, method: 'a', methodArgs: [] });
      await runMethod({ appClient, method: 'd', methodArgs: [] });
      await runMethod({ appClient, method: 'ad', methodArgs: [] });
    });
  });
});
