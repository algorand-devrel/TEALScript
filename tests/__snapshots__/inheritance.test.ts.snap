// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Inheritance A Artifacts Maintains program size 1`] = `86`;

exports[`Inheritance B Artifacts Maintains program size 1`] = `116`;

exports[`Inheritance C Artifacts Maintains program size 1`] = `143`;

exports[`Inheritance D Artifacts Maintains program size 1`] = `85`;

exports[`Inheritance AD Artifacts Maintains program size 1`] = `138`;

exports[`Inheritance E Artifacts Maintains program size 1`] = `205`;
